.github/workflows/publish-docfx.yml

```diff
@@ -6,6 +6,9 @@ on:
   push:
     branches: ["main"]
 
+  pull_request:
```

Comment:
why do we need to trigger on PR events?

I think the only event that should trigger the deployment of the doc is push on main with a file filter set to the docs folder.

Reply:
Agreed, my initial reasoning was having this workflow run on PR would help reviewers see UI changes, but we could just include them in the description. I can remove this.